internal/config/config.go

```diff
@@ -58,6 +63,35 @@ type kvConfig struct {
 	Port     uint   `json:"port,omitempty"`
 }
 
+type InternalCAConfig struct {
+	Cert		string	`json:"cert"`
+	Key        	string	`json:"key"`
+	Serial		string	`json:"serial"`
+	SignerName	string	`json:"signername"`
+	ExpireDays	uint	`json:"expiry"`
+}
+
+type HashiCAConfig struct {
+	AppRole		string	`json:"approle"`
+	SecretID	string	`json:"secretid"`
+	CAURL		string	`json:"caUrl"`
+	AppRoleURL	string	`json:"approleUrl"`
+	Signer		string	`json:"signer"`
+	Role		string	`json:"role"`
+}
+
+const (
```

Comment:
enum?

type CAType int
const (
	InternalCA CAType = iota + 1
	HashiVault
)

Reply:
Sure. Will make it an enum in next revision